task/test-cache

```diff
@@ -55,26 +55,26 @@ class TestCache(unittest.TestCase):
         self.assertEqual(result, other)
 
     def testCurrent(self):
-        c = cache.Cache(self.directory, lag=3)
+        c = cache.Cache(self.directory, lag=0.5)
```

Comment:
I don't like this test, and I particularly don't like your changes to it... this timing-dependent stuff is asking for trouble, and making the window smaller makes it more dangerous.

Why not introduce a better separation between linting and unit tests?

Reply:
I ran the test ~ 10 times in a row and that seemed fine, this is mostly a pre-push hook annoyance as that takes 6 seconds. Hmm can we have a different approach of testing this.

Reply:
Dropped it, as I don't feel like bikeshedding in this PR, we can discuss it separate from this PR :)